music_assistant/controllers/player_queues.py

```diff
@@ -453,14 +475,21 @@ async def play_media(
             queue.radio_source += radio_source
         # Use collected media items to calculate the radio if radio mode is on
         if radio_mode:
-            media_items = await self._get_radio_tracks(
+            radio_tracks = await self._get_radio_tracks(
                 queue_id=queue_id, is_initial_radio_mode=True
             )
+            media_items = list(radio_tracks)
 
         # only add valid/available items
-        queue_items = [
-            QueueItem.from_media_item(queue_id, x) for x in media_items if x and x.available
-        ]
+        # _resolve_media_items should only return playable types, but we filter to be safe
+        queue_items: list[QueueItem] = []
+        for x in media_items:
+            if not x or not x.available:
+                continue
+            # Only playable media types can be added to queue
+            if not isinstance(x, (Track, Radio, Audiobook, PodcastEpisode)):
```

Comment:
I think this is an unwanted side effect and we should rely on _resolve_media_items to only provide us with playable items. Now we basically put this responsibility in two places.